src/main/app/pay/payment.ts

```diff
@@ -1,37 +1,29 @@
-export class PaymentState {
-  status: string
-  finished: boolean
+/* tslint:disable variable-name allow snake_case */
 
-  deserialize (input?: any): PaymentState {
-    if (input) {
-      this.status = input.status
-      this.finished = input.finished
-    }
-    return this
-  }
+import 'reflect-metadata'
+import { plainToClass, Type } from 'class-transformer'
+
+class Link {
+  readonly href: string
+  readonly method: string
+}
+
+class Links {
+  @Type(() => Link)
```

Comment:
Is `@Type` needed here and below? As far as I understand the [documentation](https://github.com/typestack/class-transformer#working-with-nested-objects) it's needed only when you're transforming nested objects, e.g. elements of an array.